app/assets/stylesheets/shared/form.scss

```diff
@@ -97,14 +81,61 @@
 
   .form-control {
     color: #000;
-    border: none;
+    border: $input-border-width $border-style $input-border-color;
 
-    &:focus {
+    &.is-invalid {
+      border-color: $red;
+    }
+
+    /*&:focus {
       box-shadow: none;
       border-color: transparent;
+    }*/
+  }
+
+  // changed field stylez
+  .form-check-input.is-changed:not(.is-invalid),
+  .form-control.is-changed:not(.is-invalid),
+  .form-select.is-changed:not(.is-invalid) {
+    border-color: $orange;
+
+    &:focus {
+      box-shadow: 0 0 $input-btn-focus-blur $input-btn-focus-width rgba($orange, $input-btn-focus-color-opacity);
+    }
+
+    ~ label {
+      color: $orange;
+    }
+  }
+  .btn-discard-changed {
+    border-color: $orange;
+    &:hover {
+      border-color: $orange;
     }
   }
 
+  // bootstrap5-tags need help
+  .form-floating {
+    > .form-tag-select ~ label {
+      opacity: $form-floating-label-opacity !important;
+      transform: $form-floating-label-transform !important;
+    }
+    > .form-tag-select.form-control-blank:not(.form-control-focus) ~ label {
+      opacity: 1 !important;
+      transform: none !important;
+    }
+  }
+  .is-changed ~ .form-tag-select {
+    border-color: $orange;
+  }
+  .form-tag-select.form-control-focus {
+    color: $input-focus-color;
+    background-color: $input-focus-bg;
+    border-color: $input-focus-border-color;
+    outline: 0;
+    box-shadow: $input-focus-box-shadow;
+  }
```

Comment:
Still seeing some issues on the episode -> categories field.  When going from an initial blank state to adding the 1st tag.  But... that can be cleaned up later!  (I think that field still needs some non-UI changes too).